app/models/course/material/folder.rb

```diff
@@ -3,6 +3,8 @@ class Course::Material::Folder < ActiveRecord::Base
   include Course::ModelComponentHost::Component
 
   after_initialize :set_defaults, if: :new_record?
+  before_validation :assign_valid_name
+  before_save :assign_valid_name
```

Comment:
Yes, like this case: https://github.com/Coursemology/coursemology2/blob/master/app/models/course/assessment/category.rb#L16

If you build a folder in `before_create`, and save the parent, then folder's `before_validation` callback won't be called.